nuxeo-features/nuxeo-platform-audit/nuxeo-platform-audit-api/src/main/java/org/nuxeo/ecm/platform/audit/api/document/DocumentAuditHelper.java

```diff
@@ -87,11 +86,64 @@ public static AdditionalDocumentAuditParams getAuditParamsForUUID(String uuid, C
                 return null;
             }
             result = new AdditionalDocumentAuditParams();
-            Calendar estimatedDate = ((Calendar) doc.getPropertyValue("dc:modified"));
 
-            // We can not directly use the repo timestamp because Audit and VCS can be in separated DB
-            // => try to find the matching TS in Audit
-            StringBuilder queryString = new StringBuilder();
+            List<LogEntry> dateEntries = runNativeQueryForBackend(session, doc, targetUUID, reader);
+            if (dateEntries.size() > 0) {
+                result.targetUUID = targetUUID;
+                Calendar maxDate = new GregorianCalendar();
+                maxDate.setTime(dateEntries.get(0).getEventDate());
+                maxDate.add(Calendar.MILLISECOND, -500);
+                result.maxDate = maxDate.getTime();
+            } else {
+                // no other choice : use the VCS TS
+                // results may be truncated in some DB config
+                result.targetUUID = targetUUID;
+                result.maxDate = ((Calendar) doc.getPropertyValue("dc:modified")).getTime();
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Fix to generate and run the query based on the backend available for the Audit.
+     *
+     * @return
+     * @since 8.10
+     */
+    @SuppressWarnings("unchecked")
+    protected static List<LogEntry> runNativeQueryForBackend(CoreSession session, DocumentModel doc, String targetUUID,
+            AuditReader reader) {
+        Calendar estimatedDate = ((Calendar) doc.getPropertyValue("dc:modified"));
+
+        // We can not directly use the repo timestamp because Audit and VCS can be in separated DB
+        // => try to find the matching TS in Audit
+        Map<String, Object> params = new HashMap<String, Object>();
+        params.put("minDate", estimatedDate.getTime());
+        estimatedDate.add(Calendar.MILLISECOND, -500);
+
+        StringBuilder queryString = new StringBuilder();
+        // Check if the audit uses elasticsearch as backend
+        boolean isESBackEnd = Framework.isBooleanPropertyTrue("audit.elasticsearch.enabled");
```

Comment:
I'd rather not read the property but instead check the backend itself. So

    boolean isESBackEnd = reader.getClass().getSimpleName().equals("ESAuditBackend");

which I know is dirty but this is for a backport and it will be cleaned up in the master version by adding a proper method on `AuditReader`.